File: 35-SearchInsertPosition/35-SearchInsertPosition.c
// Last updated: 4/9/2026, 11:12:47 AM
int searchInsert(int* nums, int numsSize, int target) {
    int left = 0;
    int right = numsSize-1;

    while(left<=right){
        int mid = (left+right)/2;

        if(nums[mid]==target){
            return mid;
        }

        else if(nums[mid]<target){
            left +=1;
        }
        else{
            right -=1;
        }
    }

    return left;
}

File: 35-SearchInsertPosition/35-SearchInsertPosition.js
// Last updated: 4/9/2026, 11:12:53 AM
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let left = 0;
    let right = nums.length-1;

    while(left<=right){
        let mid = Math.floor((left+right)/2);

        if(nums[mid]===target){
            return mid;
        }
        else if(nums[mid]<target){
           left = left+1;
        }
        else{
            right = right - 1;
        }
    }

    return left;
};